kits/arm/example_impedance_control_cartesian.cpp

```diff
@@ -0,0 +1,187 @@
+  Eigen::VectorXd pos_nan(num_joints), vel_nan(num_joints);
+  pos_nan.fill(std::numeric_limits<double>::quiet_NaN());
+  vel_nan.fill(std::numeric_limits<double>::quiet_NaN());
+
+  // Add the plugin to the arm
+  if (!arm->addPlugin(std::move(impedance_plugin))) {
+    std::cerr << "Failed to add ImpedanceController plugin to arm." << std::endl;
+    return -1;
+  }
+
+  //////////////////////////
+  //// MobileIO Setup //////
+  //////////////////////////
+
+  // Set up MobileIO
+  std::unique_ptr<util::MobileIO> mobile = util::MobileIO::create(params.families_[0], "mobileIO");
+  if (!mobile)
+  {
+    std::cout << "couldn't find mobile IO device!\n";
+    return 1;
+  }
+  mobile->setButtonMode(1, util::MobileIO::ButtonMode::Momentary);
+  mobile->setButtonLabel(1, "❌");
+  mobile->setButtonMode(2, util::MobileIO::ButtonMode::Toggle);
+  mobile->setButtonLabel(2, "💪");
+
+  std::string instructions;
+  instructions = "                    Cartesian demo";
+  
+  // Clear any garbage on screen
+  mobile->clearText(); 
+
+  // Display instructions on screen
+  mobile->appendText(instructions); 
+
+  // Setup instructions
+  auto last_state = mobile->update();
+
+  std::cout <<  "Commanded gravity-compensated zero force to the arm.\n"
+            <<  "  💪 (B2) - Toggles an impedance controller on/off:\n"
+            <<  "            ON  - Apply controller based on current position\n"
+            <<  "            OFF - Go back to gravity-compensated mode\n"
+            <<  "  ❌ (B1) - Exits the demo.\n";
+
+  /////////////////////////////
+  // Control Variables Setup //
+  /////////////////////////////
+
+  // Flag to indicate when impedance controller is on
+  bool controller_on = false;
+
+  //////////////////////////
+  //// Main Control Loop ///
+  //////////////////////////
+
+  while(arm->update())
+  {
+    auto updated_mobile = mobile->update(0);
+
+    if (updated_mobile)
+    {
+      /////////////////
+      // Button Presses
+      /////////////////
+
+      // Buttton B1 - End demo
+      if (mobile->getButtonDiff(1) == util::MobileIO::ButtonState::ToOn) {
+        // Clear MobileIO text
+        mobile->resetUI();
+        return 1;
+      }
+
+      // Button B2 - Set and unset impedance mode when button is pressed and released, respectively
+      if (mobile->getButtonDiff(2) == util::MobileIO::ButtonState::ToOn) {
+
+        controller_on = true;
+
+        arm->setGoal(arm::Goal::createFromPosition(arm->lastFeedback().getPosition()));
+      }
+      else if (mobile->getButtonDiff(2) == util::MobileIO::ButtonState::ToOff){
+
+        controller_on = false;
+      }
+    }
+
+    if (!controller_on)
+    {
+      arm->cancelGoal();
+    }
+
+    // Clear all position and velocity commands
+    command.setPosition(pos_nan);
+    command.setVelocity(vel_nan);
+
+    // Send latest commands to the arm
+    arm->send();
+  }
+
+  // Clear MobileIO text
+  mobile->clearText();
```

Comment:
Interesting -- it looks like we quit if the arm->update call fails due to, e.g., a connection error?  This isn't necessarily a bad approach, because it can keep the impedance controller from getting unstable on bad connections, but we might want to signal why this happened with a printout or mobile IO message so it is clear to the user (and demo-code-reader!)